### ISSUE TITLE
[Backend] Creating Basic Admin Approval Mechanism 

### ISSUE THREAD

**Issue by ahmetkudu** [bounswe/bounswe2023group6#524]
### Problem

When a user creates s game, this game needs to be approved by admin to After the confirmation, game will be appearing everyone. Thus, user won't be able to create games that have wrong information or that they created before.

### Documentation

In this issue

### Reviewers

Ömer Bahadıroğlu

### Deadline

12.12.2023 - 23.59

**Comment by ahmetkudu** [comment]
Added a code providing an admin user when the program is started. As far as I searched, this is the best practice to create an admin on websites, like putting the initiating code and helping the system create the admin user to the source code. 
![Screenshot 2023-12-12 at 08 38 09](https://github.com/bounswe/bounswe2023group6/assets/74921213/3fffd71a-6ab2-4d73-b3c9-6fefc394feeb)

Added a field in the user entity as IsAdmin. I didn't put the IsAdmin field to the UserDTO, especially in the request and response one, because I don't want users to create an admin user using the related endpoint since this would create an insecure system.
![Screenshot 2023-12-12 at 08 38 24](https://github.com/bounswe/bounswe2023group6/assets/74921213/ec4ae419-6bc6-4502-b1ce-873b45721539)

This is the endpoint needed to manage admin tasks: viewing all pending games and taking action about games that will be approved or rejected.
![Screenshot 2023-12-12 at 08 42 52](https://github.com/bounswe/bounswe2023group6/assets/74921213/dc012653-c4fc-4882-a6e3-c9ddd60bf6bd)

This is the main code doing the job in the game service:
![Screenshot 2023-12-12 at 08 54 51](https://github.com/bounswe/bounswe2023group6/assets/74921213/8183c86d-cfda-449f-8070-52c354e376a4)
